src/jinja2/ext.py

```diff
@@ -456,11 +457,25 @@ def parse(self, parser):
 
 
 class WithExtension(Extension):
-    pass
+    def __init__(self, environment):
+        Extension.__init__(self, environment)
+        warnings.warn(
+            "WithExtension is deprecated and will be removed in version 3.1",
+            DeprecationWarning,
+            stacklevel=2,
+        )
 
 
 class AutoEscapeExtension(Extension):
-    pass
+    def __init__(self, environment):
+        Extension.__init__(self, environment)
+        warnings.warn(
+            "This extension is deprecated and will be removed in version 3.1",
```

Comment:
`The 'autoescape' extension is deprecated and will be removed in version 3.1. This is built-in now.`